index.rst

```diff
@@ -1,57 +1,236 @@
+
+The four LSST Middleware interactions needed by the framework are:
+
+ * On startup, initialize a handler-local Butler repo and copy globally-used calibrations.
+ * On receipt of ``next_visit``, copy calibrations, templates, and potentially APDB data required by the pipeline into the handler-local repo.
+ * On receipt of an image, ingest its metadata.
+ * On receipt of the last image in a visit, trigger execution of the selected pipeline.
+
+Creating a handler-local repo is necessary in order to execute unmodified pipelines, without requiring the writing of special code to work within Prompt Processing.
+This repo (which is expected to be located on a RAM disk or fast local SSD) is used to ensure that all information needed by the pipeline is preloaded to the worker, avoiding bottlenecks during the critical time period after images have been received.
+The code to preload data could be custom or it could be written as PipelineTasks and executed as an "initial step" pipeline, writing outputs to the handler-local repo.
+
+It is assumed that all information needed to preload data and to identify the images to be processed is present in the ``next_visit`` event.
+In particular, this means that the message handler must be able to know how many snap exposures are going to be received and what their URIs will be.
+It is not feasible to poll a central repo to determine what images have arrived and been ingested, as this adds substantial latency, both for the central repo triggering and ingest and for the poll and copy to the handler-local repo.
+A blocking-with-callback or other asynchronous API to the Butler could reduce the second source of latency, but it would not affect the first.
+
+Copying the image from its publication point to the handler-local repo is assumed to be an irreducible cost.
+By placing the image in permanent storage as soon as possible after retrieval from the Camera Data Acquisition system, the possibility of loss is minimized.
+Previously, Prompt Processing was to receive a special, low-latency copy of the image (with firmware-based crosstalk correction), while the permanent archive would receive a non-crosstalk-corrected version after some delay (after the network burst caused by Prompt Processing).
+Since the Camera is no longer doing firmware crosstalk correction, moving the copy to the destination end of the international link avoids congesting that link, places it in a high-bandwidth local network environment, and enables all potential USDF users of the images to receive the same low-latency service.
+
+The Butler interactions are handled via the Python API for export and import, and, for simplicity of invocation, the command line tooling for ingest and pipeline execution.
+
+Using a batch queue mechanism for firing off the pipeline execution is attractive in that the HTTP handler would not need to persist for the lifetime of the pipeline, and resilience and elasticity could be built into the batch system.
```

Comment:
This contradicts section 2:
> The HTTP message handler does not return until the visit has been successfully processed or a failure has occurred, been identified, and reported.

Reply:
I'm saying that _if_ we had a batch queue we could potentially hand off to it, allowing the HTTP message handler to return before the pipeline has completed.  Of course the handler status would then be less meaningful, so status would have to be monitored via a different mechanism.